src/MongoDB/Manager.c

```diff
@@ -669,22 +671,53 @@ static PHP_METHOD(Manager, startSession)
 		return;
 	}
 
-	if (options && php_array_exists(options, "causalConsistency")) {
+	if (options && php_array_existsc(options, "causalConsistency")) {
 		cs_opts = mongoc_session_opts_new();
 		mongoc_session_opts_set_causal_consistency(cs_opts, php_array_fetchc_bool(options, "causalConsistency"));
 	}
 
-	cs = mongoc_client_start_session(intern->client, cs_opts, &error);
+	if (options && php_array_existsc(options, "defaultTransactionOptions")) {
+		zval* txn_options = php_array_fetchc(options, "defaultTransactionOptions");
 
-	if (cs_opts) {
-		mongoc_session_opts_destroy(cs_opts);
+		/* Thrown exception and return if the defaultTransactionOptions is not an array */
+		if (Z_TYPE_P(txn_options) != IS_ARRAY) {
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected \"defaultTransactionOptions\" option to be an array");
```

Comment:
I think my original request in https://github.com/mongodb/mongo-php-driver/pull/850#discussion_r195877062 was lost by subsequent edits. Can we add ", %s given" for consistency with other exceptions? The inconsistency stands out in the error tests you have below, which is also how I realized this change wasn't made.

Reply:
Done